src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java

```diff
@@ -742,6 +742,11 @@ public Integer getProjectIdByStudyDbId(final Integer studyDbId) {
 		return this.getDmsProjectDao().getProjectIdByStudyDbId(studyDbId);
 	}
 
+	@Override
+	public Integer getDatasetIdByStudyDbIdAndDatasetType(final Integer studyDbId, final DatasetTypeEnum datasetType) {
```

Comment:
StudyDbId is a BrAPI concept. 
In light of IBP-3305 (removal of nd_geolocation in experiments), we should use "environment id" as common term in Middleware. Apply for all parameter, variables etc